src/core/dom.ts

```diff
@@ -122,11 +126,19 @@ export class AbstractWrapper<ElementType extends Element>
 
   findComponent<Wrapper extends ComponentWrapper<ElementType>, ElementType extends HTMLElement>(
     selector: string,
-    ComponentClass: { new (element: ElementType): Wrapper }
+    ComponentClass: WrapperClass<Wrapper, ElementType>
   ): Wrapper | null {
     const elementWrapper = this.find<ElementType>(selector);
     return elementWrapper ? new ComponentClass(elementWrapper.getElement()) : null;
   }
+
+  findAllComponents<Wrapper extends ComponentWrapper<ElementType>, ElementType extends HTMLElement>(
+    selector: string,
+    ComponentClass: WrapperClass<Wrapper, ElementType>
+  ): Array<Wrapper> {
+    const elementWrappers = this.findAll<ElementType>(selector);
+    return elementWrappers.map(wrapper => new ComponentClass(wrapper.getElement()));
+  }
```

Comment:
This function takes any selector and force casts its return into `ComponentClass` type, even if the selector doesn't point to a `ComponentClass` type.

I don't think this should be the behavior but since this is what we already have with `findComponent` I followed the same approach here as well.

The correct behavior IMO should be:
```tsx
findAllComponents(ComponentClass, selector /* optional */) {
  const elementWrappers = this.findAll<ElementType>(appendSelector(ComponentClass.rootSelector, selector));
  ...
}
```

This won't return anything if it's not the intended component type.

Reply:
Nice finding. I would be more inclined to take this opportunity to do this right for the new API, and later see if we can and want to fix it for the old one —but first can we make sure there was not a reason to do it like that?